src/hooks/UrqlUseMutation.re

```diff
@@ -41,28 +41,33 @@ let urqlResponseToReason =
  *
  * Accepts the following arguments:
  *
- * request – a Js.t containing the query and variables corresponding
- * to the GraphQL mutation, and a parse function for decoding the JSON response.
+ * request – a tuple containing the query, a parse function for decoding the
+ * JSON response, and a higher-order function that calls a passed-in function
+ * with the variables corresponding to the GraphQL. This should be generated by
+ * graphql_ppx_re and is accessible in the MyQuery.definition variable.
  */;
-let useMutation = (~request) => {
-  let (responseJs, executeMutationJs) = useMutationJs(request##query);
+let useMutation = ((parse, query, composeVariables)) => {
+  composeVariables(variables => {
+    let (responseJs, executeMutationJs) = useMutationJs(query);
 
-  let response =
-    React.useMemo2(
-      () => responseJs |> urqlResponseToReason(request##parse),
-      (request##parse, responseJs),
-    );
+    let response =
+      React.useMemo2(
+        () => responseJs |> urqlResponseToReason(parse),
+        (parse, responseJs),
+      );
 
-  let executeMutation =
-    React.useCallback1(
-      () => executeMutationJs(Some(request##variables)),
-      [|request##variables|],
-    );
+    let executeMutation =
+      React.useCallback1(
+        () => executeMutationJs(Some(variables)),
+        [|variables|],
+      );
```

Comment:
A question here that I'm sure we solved at some point but for some reason is slipping my mind – what about mutations that don't accept any variables? It seems like this impl. (and our old one too) assume that you have `variables`. Does `graphql_ppx_re` reliably return no `variables` as something like `Js.Json.object_(Js.Dict.empty())`? Just wanting to make sure we handle the no variable mutation case which, while rare, does exist.